bridgebot/cards.py

```diff
@@ -0,0 +1,171 @@
+import random
+
+from enums import Suits, Ranks
+
+
+class InvalidCardException(Exception):
+    pass
+
+
+class InvalidRankException(InvalidCardException):
+    pass
+
+
+class InvalidSuitException(InvalidCardException):
+    pass
+
+
+class InvalidIndexException(Exception):
+    pass
+
+
+class CardNotInHandException(Exception):
+    pass
+
+
+class CardDoesntFollowSuitException(Exception):
+    pass
+
+
+class WrongSizeHandException(Exception):
+    pass
+
+
+class Card:
+    @staticmethod
+    def generate_card_from_index(index):
+        if index not in range(52):
+            raise InvalidIndexException("Index must be an integer between 0 and 51 inclusive.")
+        return Card(Suits.suits()[int(index / 13)], Ranks.ranks()[index % 13])
+
+    def __init__(self, suit, rank):
+        if not isinstance(suit, Suits):
+            raise InvalidSuitException("Suit not found")
+
+        if not isinstance(rank, Ranks):
+            raise InvalidRankException("Rank not found")
+
+        self.suit = suit
+        self.rank = rank
+
+    def __eq__(self, other):
+        return self.suit == other.suit and self.rank == other.rank
+
+
+class DeckListNotValid(Exception):
+    pass
+
+
+class BridgeHand:
```

Comment:
I originally had it broken out as two classes, `Hand` and `BridgeHand` because `Hand` was reusable. Especially for cases where we want to load the game halfway through the hand, we don't want to restrict to 13 cards. For the purposes of training the AI, I think we want to be able to load more general hands. For example, training on only 5 cards to train how to deal with the endgame.

Reply:
For the AI bot, we should consider moving the state out entirely so that we can create a replayable/loadable from disk input. I think over time, we need to serialize the state of the program and be able to load it back into the game.